PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx

```diff
@@ -0,0 +1,321 @@
+                        /* Σc0,++ specific columns */
+                        chargeSigmac,
+                        statusSpreadMinvPKPiFromPDG, statusSpreadMinvPiKPFromPDG);
+
+      } /// end loop over tracks
+
+    } /// end loop over candidtes
+
+  } /// end process
+};
+
+/// Extends the base table with expression columns.
+
+struct HfCandidateSigmac0plusplusMc {
+
+  Spawns<aod::HfCandScExt> candidatesSigmac;
+  Produces<aod::HfCandScMcRec> rowMCMatchScRec;
+  Produces<aod::HfCandScMcGen> rowMCMatchScGen;
+
+  using LambdacMc = soa::Join<aod::HfCand3Prong, aod::HfSelLc, aod::HfCand3ProngMcRec>;
+  // using LambdacMcGen = soa::Join<aod::McParticles, aod::HfCand3ProngMcGen>;
+  using TracksMC = soa::Join<aod::Tracks, aod::McTrackLabels>;
+
+  /// @brief init function
+  void init(InitContext const&) {}
+
+  /// @brief dummy process function, to be run on data
+  /// @param
+  void process(const aod::Tracks&) {}
+
+  /// @brief process function for MC matching of Σc0,++ → Λc+(→pK-π+) π- reconstructed candidates and counting of generated ones
+  /// @param candidatesSigmac reconstructed Σc0,++ candidates
+  /// @param particlesMc table of generated particles
+  void processMc(aod::McParticles const& particlesMc, const TracksMC& tracks,
+                 LambdacMc const& /*, const LambdacMcGen&*/)
+  {
+
+    // Match reconstructed candidates.
+    candidatesSigmac->bindExternalIndices(&tracks);
+
+    int indexRec = -1;
+    int8_t sign = 0;
+    int8_t flag = 0;
+    int8_t origin = 0;
+    int8_t chargeSigmac = 10;
+    // std::vector<int> arrDaughIndex; /// index of daughters of MC particle
+
+    /// Match reconstructed Σc0,++ candidates
+    for (auto const& candSigmac : *candidatesSigmac) {
+      indexRec = -1;
+      sign = 0;
+      flag = 0;
+      origin = 0;
+      // arrDaughIndex.clear();
+
+      /// skip immediately the candidate Σc0,++ w/o a Λc+ matched to MC
+      auto candLc = candSigmac.prongLc_as<LambdacMc>();
+      if (!(std::abs(candLc.flagMcMatchRec()) == 1 << DecayType::LcToPKPi)) { /// (*)
+        rowMCMatchScRec(flag, origin);
+        continue;
+      }
+
+      /// matching to MC
+      auto arrayDaughters = array{candLc.prong0_as<TracksMC>(),
+                                  candLc.prong1_as<TracksMC>(),
+                                  candLc.prong2_as<TracksMC>(),
+                                  candSigmac.prong1_as<TracksMC>()};
+      chargeSigmac = candSigmac.charge();
+      if (chargeSigmac == 0) {
+        /// candidate Σc0
+        /// 3 levels:
+        ///   1. Σc0 → Λc+ π-,+
+        ///   2. Λc+ → pK-π+ direct (i) or Λc+ → resonant channel Λc± → p± K*, Λc± → Δ(1232)±± K∓ or Λc± → Λ(1520) π±  (ii)
+        ///   3. in case of (ii): resonant channel to pK-π+
+        indexRec = RecoDecay::getMatchedMCRec(particlesMc, arrayDaughters, pdg::Code::kSigmaC0, array{static_cast<int>(kProton), static_cast<int>(kKMinus), static_cast<int>(kPiPlus), static_cast<int>(kPiMinus)}, true, &sign, 3);
```

Comment:
`static_cast<int>` should not be needed. It works fine without it in the other candidate creators.

Reply:
I changed as done in other tasks.